### PR TITLE
Make xrepl the default repl if it is installed

### PR DESCRIPTION
**Pull request by LeifAndersen**

### CODE REVIEW

**Comment by stamourv** [comment]
:tada:
LGTM

**Comment by LeifAndersen** [comment]
Okay, will merge as soon as the CI tests pass.

File: racket/collects/racket/init.rkt [hunk 1]
```diff
@@ -12,3 +12,7 @@
 (provide (all-from-out racket
                        racket/enter
                        racket/help))
+
+;; Set the default replt to XREPL
+(with-handlers ([exn:fail? void])
```
Comment (samth) [inline racket/collects/racket/init.rkt]: Couldn't this use the test that I suggested: `(collection-file-path "main.rkt" "xrepl" #:fail (lambda _ #f))`?

Comment (LeifAndersen) [inline racket/collects/racket/init.rkt]: Yes. although FWIW, this didn't seem any better than your other suggestion, and so I went with that one. I don't really have much of a preference and would be happy to change it if you feel strongly one way or the other.

Comment (samth) [inline racket/collects/racket/init.rkt]: The major difference is that your version will swallow other errors, such as not having readline or editline available.

Comment (LeifAndersen) [inline racket/collects/racket/init.rkt]: That it will, is that a problem though?

Comment (samth) [inline racket/collects/racket/init.rkt]: Depends what you want the behavior to be.

Comment (LeifAndersen) [inline racket/collects/racket/init.rkt]: Fair. Well also iirc, xrepl actually swallows up the error itself if libedit or libreadline are not installed, and works normally, just without readline support.

Either way though, this doesn't really matter, so I'll just change it to the other way (the second one you suggested in slack using `collection-file-path` and call it good.

**Comment by LeifAndersen** [comment]
merged

**Comment by rfindler** [comment]
FWIW, I think it is better not to swallow errors. (cf: my dissertation and lots of pain over the years in the DrRacket REPL implementation)

**Pull request closed without merging**